conans/client/conan_api.py

```diff
@@ -173,19 +173,22 @@ def instance_remote_manager(requester, cache, user_io, hook_manager):
     def factory(interactive=None):
         """Factory"""
         # Respect color env setting or check tty if unset
-        color_set = "CONAN_COLOR_DISPLAY" in os.environ
-        if ((color_set and get_env("CONAN_COLOR_DISPLAY", 1))
-                or (not color_set
-                    and hasattr(sys.stdout, "isatty")
-                    and sys.stdout.isatty())):
+        for var in ("CONAN_COLOR_DISPLAY", "PYCHARM_HOSTED"):
+            if var in os.environ:
+                force_color = get_env(var, False)
+                color = force_color
+                break
+        else:
+            force_color = False
+            color = hasattr(sys.stdout, "isatty") and sys.stdout.isatty()
+
+        if color:
             import colorama
-            if get_env("PYCHARM_HOSTED"):  # in PyCharm disable convert/strip
+            if force_color:
                 colorama.init(convert=False, strip=False)
```

Comment:
Previously this was only done when `PYCHARM_HOSTED` but now also with `CONAN_COLOR_DISPLAY`, right? I would need some explanation about this change and how it fixes the issue.

Reply:
That is the fix: if user sets `CONAN_COLOR_DISPLAY` to `1` it should behave exactly as when `PYCHARM_HOSTED` present. (Research in my comment to the issue https://github.com/conan-io/conan/issues/4718#issuecomment-478533897)

Major complaint is that setting `CONAN_COLOR_DISPLAY=1` doesn't really force colors (as I as a user would expect) because colorama does check for tty internally and essentially this variable is being ignored.

Now when either of those variables are set to 1 conan forces ANSI sequences in the output regardless whether it's a tty or not. When it's set to 0 the effect is the opposite: no colors even when stdout is a tty.
That's why 2 variables: `force_color` and `color`. If `force_color` is not set then colorama is initialized as before with default parameters.